WearOAuth/oauth-device-grant/src/main/java/com/example/android/wearable/oauth/devicegrant/AuthDeviceGrantViewModel.kt

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.android.wearable.oauth.devicegrant
+
+import android.content.Intent
+import android.net.Uri
+import android.util.Log
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import androidx.wear.remote.interactions.RemoteIntentHelper
+import com.example.android.wearable.oauth.util.doGetRequest
+import com.example.android.wearable.oauth.util.doPostRequest
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+private const val TAG = "AuthDeviceGrantViewModel"
+
+// TODO Add your client id & secret here (for dev purposes only).
+private const val CLIENT_ID = ""
+private const val CLIENT_SECRET = ""
+
+/**
+ * The viewModel that implements the OAuth flow. The method [startAuthFlow] implements the
+ * different steps of the flow. It first retrieves the URL that should be opened on the paired
+ * device, then polls for the access token, and uses it to retrieve the user's name.
+ */
+class AuthDeviceGrantViewModel(private val remoteIntentHelper: RemoteIntentHelper) : ViewModel() {
+    // Status to show on the Wear OS display
+    val status: MutableLiveData<String> by lazy { MutableLiveData<String>() }
+    private fun showStatus(statusString: String) = status.postValue(statusString)
+
+    fun startAuthFlow() {
+        viewModelScope.launch(Dispatchers.IO) {
```

Comment:
Now that we're not calling any blocking functions directly, we probably don't need to care what thread we're on here.

Reply:
Fixed!